client/client.go

```diff
@@ -305,10 +299,13 @@ func (c *Client) downloadTimestamp() error {
 func (c *Client) downloadSnapshot() error {
```

Comment:
There is a lot of custom logic for each role that determines things like where (and whether) a size and sha256 are retrieved, i.e. for a snapshot, we get them from the timestamp, and they _must_ be present, where as for a timestamp, we don't have them at all, and for targets/root, we get them from the snapshot, and again, they must be present.

I haven't worked out how/if it's possible to unify more of the role based code but that's what I'm trying to move towards with things like `downloadSigned`